File: Top 100 Codes/01-Getting Started/14-Armstrong number/Armstrongnumber.py
# Function to check if a number is an Armstrong number
def is_armstrong(number):
    """
    Checks whether the given number is an Armstrong number.
    
    An Armstrong number (or Narcissistic number) is a number that is equal to
    the sum of its own digits raised to the power of the number of digits.
    
    Args:
        number (int): The number to check.
    
    Returns:
        bool: True if the number is an Armstrong number, False otherwise.
    """
    # Convert the number to a string to calculate the number of digits
    num_str = str(number)
    num_digits = len(num_str)
    
    # Calculate the sum of the digits raised to the power of num_digits
    armstrong_sum = sum(int(digit) ** num_digits for digit in num_str)
    
    # Check if the sum is equal to the original number
    return armstrong_sum == number

# Prompt the user to enter a number
num = int(input("Enter a number: "))

# Check if the input number is an Armstrong number and display the result
if is_armstrong(num):
    print(f"{num} is an Armstrong number.")
else:
    print(f"{num} is not an Armstrong number.")

File: Top 100 Codes/01-Getting Started/14-Armstrong number/Armstrongnumber.cpp
#include <iostream>
#include <cmath>
using namespace std;

// Function to check if a number is an Armstrong number
bool isArmstrong(int num) {
    int originalNum = num; // Store the original number
    int sum = 0;
    int digits = 0;

    // Calculate the number of digits
    int temp = num;
    while (temp != 0) {
        temp /= 10;
        digits++;
    }

    // Calculate the sum of each digit raised to the power of the number of digits
    temp = num;
    while (temp != 0) {
        int rem = temp % 10;         // Extract the last digit
        sum += pow(rem, digits);    // Add the power of the digit to the sum
        temp /= 10;                 // Remove the last digit
    }

    // Check if the sum is equal to the original number
    return sum == originalNum;
}

int main() {
    int num;

    // Prompt the user for input
    cout << "Enter a number: ";
    cin >> num;

    // Check if the number is an Armstrong number and display the result
    if (isArmstrong(num)) {
        cout << num << " is an Armstrong number." << endl;
    } else {
        cout << num << " is not an Armstrong number." << endl;
    }

    return 0;
}
